solidity/contracts/Gravity.sol

```diff
@@ -180,6 +191,31 @@ contract Gravity is ReentrancyGuard {
 		}
 	}
 
+	// Utility function to check for duplicate validators
+	// The checks are performed in memory and discarded
+	function checkDuplicateValidators(address[] memory _validators) private {
+		ValidatorPosition nextPosition;
+
+		if (
+			currentPosition == ValidatorPosition.UNINITIALIZED ||
+			nextPosition == ValidatorPosition.RIGHT
+		) {
+			nextPosition = ValidatorPosition.LEFT;
+		} else {
+			nextPosition = ValidatorPosition.RIGHT;
+		}
+
+		for (uint256 i = 0; i < _validators.length; i++) {
+			if (validatorSet[_validators[i]] == nextPosition) {
+				revert MalformedNewValidatorSet();
+			} else {
+				validatorSet[_validators[i]] = nextPosition;
```

Comment:
Can just reset the mapping when we finish? For example we ca use the ```mapping(unit  => mapping(address => ValidatorPosition))``` and use constant value "0" for the validator set and a unit key. And then, at the end just reset the mapping with that "0" key. It would also simplify the logic with "ValidatorPosition.UNINITIALIZED" since it will be always "ValidatorPosition.UNINITIALIZED" means you can get rid of enum and use boolean instead.